src/wamr/CMakeLists.txt

```diff
@@ -1,90 +1,86 @@
-# Unfortunately the pre-canned WAMR CMake file modifies too many top-level
-# variables (e.g. through use of include_directories or changing global
-# CFLAGS).
-#
-# We have our own custom CMake set-up which looks a bit like the wamrc build:
-# https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/wamr-compiler/CMakeLists.txt
+# WARNING: the CMake flags and pre-processor definitions that we set in this
+# file are carefully chosen for the WAMR build to run. Adding or removing flags
+# may break Faasm.
 
-# Turn on WAMR features
-add_definitions(-DWASM_ENABLE_BULK_MEMORY=1)
-add_definitions(-DWASM_ENABLE_WAMR_COMPILER=1)
-add_definitions(-DWASM_ENABLE_MULTI_MODULE=1)
+# Set target platform details
+set (WAMR_BUILD_PLATFORM "linux")
+set (WAMR_BUILD_TARGET X86_64)
+set (WAMR_BUILD_SPEC_TEST 0)
 
-# Here we include all the components of WAMR we want
-include(${WAMR_SHARED_DIR}/platform/linux/shared_platform.cmake)
-include(${WAMR_SHARED_DIR}/mem-alloc/mem_alloc.cmake)
-include(${WAMR_SHARED_DIR}/utils/shared_utils.cmake)
-include(${WAMR_SHARED_DIR}/utils/uncommon/shared_uncommon.cmake)
-include(${WAMR_IWASM_DIR}/common/iwasm_common.cmake)
-include(${WAMR_IWASM_DIR}/libraries/libc-wasi/libc_wasi.cmake)
+# Set AOT mode and JIT for code generation
+set (WAMR_BUILD_AOT 1)
+set (WAMR_BUILD_JIT 1)
+set (WAMR_BUILD_LAZY_JIT 0)
 
-# enable AOT
-include(${WAMR_IWASM_DIR}/aot/iwasm_aot.cmake)
-include(${WAMR_IWASM_DIR}/interpreter/iwasm_interp.cmake)
-include(${WAMR_IWASM_DIR}/compilation/iwasm_compl.cmake)
-faasm_private_lib(aotclib "${IWASM_COMPL_SOURCE}")
+# Set libraries
+set (WAMR_BUILD_LIBC_BUILTIN 1)
+set (WAMR_BUILD_LIBC_WASI 1)
+set (WAMR_BUILD_LIB_PTHREAD 0)
 
-set(COMMON_WAMR_SRC
-        ${PLATFORM_SHARED_SOURCE}
-        ${MEM_ALLOC_SHARED_SOURCE}
-        ${IWASM_COMMON_SOURCE}
-        ${UTILS_SHARED_SOURCE}
-        ${IWASM_INTERP_SOURCE}
-        ${IWASM_AOT_SOURCE}
-        ${LIBC_WASI_SOURCE}
-)
+# WAMR features
+set (WAMR_BUILD_SIMD 0)
+
+set (WAMR_BUILD_MULTI_MODULE 1)
+# If bulk memory set, WAMR allocates one single page, breaking our brk memory
+# management model
+set (WAMR_BUILD_BULK_MEMORY 0)
 
-# Build the WAMR lib
-set(ALL_WAMR_SRC ${COMMON_WAMR_SRC} ${EXTRA_WAMR_SRC})
-faasm_private_lib(wamrlib "${ALL_WAMR_SRC}")
+# Explicitely set to 0. If set to 1, WAMR will allocate the module's instance
+# memory using `malloc` instead of `mmap` and calls to `mprotect` fail.
+set (WAMR_DISABLE_HW_BOUND_CHECK 0)
+# This definition prevents a buffer underflow during code generation
+add_definitions(-DWASM_ENABLE_WAMR_COMPILER=1)
+
+include (${WAMR_ROOT_DIR}/build-scripts/runtime_lib.cmake)
+# 13/09/2021 - Setting the CMake flag includes the code required to run
```

Comment:
The criteria for having a date here and nowhere else, is that the `remove_definitions` bit is a patch, and the other are configuration flags.

Nonetheless, I also add for each configuration flag what happens if set otherwise. Strikingly enough, most flags will break the build if set to the opposite value.